Build fails on aarch64

The build of libnv currently fails on aarch64-unknown-freebsd.  It's caused by arm using unsigned chars.
Full build log:
https://pkg-status.freebsd.org/ampere3/data/131arm64-default/d0f8db852755/logs/nfs-exporter-0.4.1.log
Relevant part:
```
error[E0277]: a value of type `Vec<*const i8>` cannot be built from an iterator over elements of type `*const u8`
   --> /wrkdirs/usr/ports/net-mgmt/nfs-exporter/work/freebsd-nfs-exporter-0.4.1/cargo-crates/libnv-0.4.1/src/libnv.rs:442:44
    |
442 |             let pointers: Vec<*const i8> = strings.iter().map(|e| e.as_ptr()).collect();
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ------- required by a bound introduced by this call
    |                                            |
    |                                            value of type `Vec<*const i8>` cannot be built from `std::iter::Iterator<Item=*const u8>`
    |
    = help: the trait `FromIterator<*const u8>` is not implemented for `Vec<*const i8>`
    = help: the trait `FromIterator<T>` is implemented for `Vec<T>`
note: required by a bound in `collect`
   --> /wrkdirs/usr/ports/lang/rust/work/rustc-1.67.1-src/library/core/src/iter/traits/iterator.rs:1832:19
    |
    = note: required by this bound in `collect`

error[E0308]: mismatched types
   --> /wrkdirs/usr/ports/net-mgmt/nfs-exporter/work/freebsd-nfs-exporter-0.4.1/cargo-crates/libnv-0.4.1/src/libnv.rs:447:17
    |
444 |             nvlist_add_string_array(
    |             ----------------------- arguments to this function are incorrect
...
447 |                 pointers.as_slice().as_ptr(),
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const *const u8`
               found raw pointer `*const *const i8`
note: function defined here
   --> /wrkdirs/usr/ports/net-mgmt/nfs-exporter/work/target/aarch64-unknown-freebsd/release/build/libnv-sys-b2def130c3866b90/out/bindings.rs:218:12
    |
218 |     pub fn nvlist_add_string_array(
    |            ^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /wrkdirs/usr/ports/net-mgmt/nfs-exporter/work/freebsd-nfs-exporter-0.4.1/cargo-crates/libnv-0.4.1/src/libnv.rs:721:47
    |
721 |                     .map(|ptr| CStr::from_ptr(ptr))
    |                                -------------- ^^^ expected `u8`, found `i8`
    |                                |
    |                                arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`
note: associated function defined here
   --> /wrkdirs/usr/ports/lang/rust/work/rustc-1.67.1-src/library/core/src/ffi/c_str.rs:256:25

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `libnv` due to 3 previous errors
```

--- a/libnv-sys/src/lib.rs
+++ b/libnv-sys/src/lib.rs
@@ -10,9 +10,6 @@
 #![allow(non_camel_case_types)]
 
 #[cfg(not(crossdocs))]
-type size_t = usize;
-
-#[cfg(not(crossdocs))]
 include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 
 #[cfg(crossdocs)]

--- a/src/libnv.rs
+++ b/src/libnv.rs
@@ -25,7 +25,7 @@ use libnv_sys::*;
 use std::{
     convert::{From, Into},
     ffi::CStr,
-    os::raw::c_void,
+    os::raw::{c_char, c_void},
     os::unix::io::AsRawFd,
     slice,
 };
@@ -439,7 +439,7 @@ impl NvList {
             .map(IntoCStr::into_c_str)
             .collect::<NvResult<Vec<_>>>()?;
         unsafe {
-            let pointers: Vec<*const i8> = strings.iter().map(|e| e.as_ptr()).collect();
+            let pointers: Vec<*const c_char> = strings.iter().map(|e| e.as_ptr()).collect();
 
             nvlist_add_string_array(
                 self.ptr,
@@ -714,13 +714,11 @@ impl NvList {
                 let mut len: usize = 0;
                 let arr =
                     nvlist_get_string_array(self.ptr, c_name.as_ptr(), &mut len as *mut usize);
-                let slice = slice::from_raw_parts(arr as *const *const i8, len);
+                let slice = slice::from_raw_parts(arr as *const *const c_char, len);
                 let strings = slice
                     .iter()
                     .copied()
-                    .map(|ptr| CStr::from_ptr(ptr))
-                    .map(|cstr| cstr.to_string_lossy())
-                    .map(String::from)
+                    .map(|ptr| String::from(CStr::from_ptr(ptr).to_string_lossy()))
                     .collect();
                 Ok(Some(strings))
             } else {

